libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortEditorFilter.tsx

```diff
@@ -42,48 +42,50 @@ export interface ICohortEditorFilterProps {
   setComparison(ev?: React.FormEvent<IComboBox>, item?: IComboBoxOption): void;
   setNumericValue(
     delta: number,
-    column: IJointMeta,
     index: number,
-    stringVal: string
+    stringVal: string,
+    range?: IColumnRange
   ): string | void;
   saveState(index?: number): void;
   cancelFilter(): void;
 }
+
 export class CohortEditorFilter extends React.Component<ICohortEditorFilterProps> {
-  private readonly dataArray: IComboBoxOption[] = new Array(
-    this.props.jointDataset.datasetFeatureCount
-  )
-    .fill(0)
-    .map((_, index) => {
-      const key = JointDataset.DataLabelRoot + index.toString();
+  private readonly dataArray: IComboBoxOption[] = this.props.featureNames.map(
+    (featureName, index) => {
       return {
-        key,
-        text: this.props.jointDataset.metaDict[key].abbridgedLabel
+        index,
+        key: featureName,
+        text:
+          featureName.length <= maxLength
+            ? featureName
+            : `${featureName.slice(0, maxLength)}...`
       };
-    });
+    }
+  );
+
   public render(): React.ReactNode {
-    const selectedMeta =
-      this.props.jointDataset.metaDict[this.props.openedFilter.column];
-    const isDataColumn = this.props.openedFilter.column.includes(
-      JointDataset.DataLabelRoot
+    const isDataColumn = this.props.featureNames.includes(
+      this.props.openedFilter.column
     );
     let categoricalOptions: IComboBoxOption[] | undefined;
     // filterIndex is set when the filter is editing openedFilter and reset to filters.length otherwise
     const isEditingFilter =
       this.props.filterIndex !== this.props.filters.length;
-    let minVal, maxVal;
-    if (selectedMeta?.treatAsCategorical || !selectedMeta.featureRange) {
-      // Numerical values treated as categorical are stored with the values in the column,
-      // true categorical values store indexes to the string values
-      categoricalOptions = selectedMeta.sortedCategoricalValues?.map(
+    const columnRange = this.props.columnRanges
+      ? this.props.columnRanges[this.props.openedFilter.column]
+      : undefined;
+    const isCategorical =
+      columnRange?.treatAsCategorical ||
+      columnRange?.rangeType === RangeTypes.Categorical;
+    if (isCategorical) {
+      categoricalOptions = columnRange?.sortedUniqueValues.map(
         (label, index) => {
-          return { key: index, text: label };
+          return { key: index, text: String(label) };
```

Comment:
is there a difference for String(A) and A.toString()?

Reply:
the type of label here is unknown, and we cannot use toString() for unknown type as it could be null or undefined. 
fyi, https://stackoverflow.com/questions/3945202/whats-the-difference-between-stringvalue-vs-value-tostring